### DISCUSSION TITLE
Best away to return updated contract offers to each consumers

### DISCUSSION THREAD

**Discussion by jannotti-glaucio** [eclipse-edc/Connector#4379] [General]
Hi guys !

I need to validate at every catalog fetch request to a consumer, which assets the consumer can access, based on your contract policies. I tried to do it using the contractDefinition assetsSelector field, but it`s a static list and I need something more dynamic, but the assets list is always updated.

How the best approach to do it ?
Implementing a (custom policy engine)[https://github.com/eclipse-edc/Connector/blob/main/docs/developer/policy-engine.md] or is there any other best way to do it ?

Regards

**Comment by jimmarino** [comment]
Please read the documentation, particularly about how access policy and selectors work on a `ContractDefinition`. The list is dynamic, based on claims presented by the client, not static.

**Comment by jannotti-glaucio** [comment]
Which documentantion ?

Em seg., 29 de jul. de 2024, 16:41, Jim Marino ***@***.***>
escreveu:

**Reply by jimmarino** [reply]
https://eclipse-edc.github.io/docs/#/README
https://github.com/eclipse-edc/Samples
https://github.com/eclipse-edc/Connector/tree/main/docs/developer

**Reply by jannotti-glaucio** [reply]
Thanks, I found what I need on this document:  https://github.com/eclipse-edc/Samples/blob/main/policy/README.md
